okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java

```diff
@@ -43,16 +46,27 @@
 import okhttp3.OkHttpClient;
 import okhttp3.TlsVersion;
 import okhttp3.internal.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public final class OkHttpClients {
+    private static final Logger log = LoggerFactory.getLogger(OkHttpClients.class);
 
     @VisibleForTesting
     static final int NUM_SCHEDULING_THREADS = 5;
 
+    private static final ThreadFactory executionThreads = new ThreadFactoryBuilder()
+            .setUncaughtExceptionHandler((thread, uncaughtException) ->
+                    log.error("An exception was uncaught in an execution thread. This implies a bug in http-remoting",
+                            uncaughtException))
+            .setNameFormat("remoting-okhttp-dispatcher-%d")
+            .build();
     /**
      * The {@link ExecutorService} used for the {@link Dispatcher}s of all OkHttp clients created through this class.
+     * Same as OkHttp's default, but with a logging uncaught exception handler.
      */
-    private static final ExecutorService executionExecutor = Tracers.wrap(new Dispatcher().executorService());
+    private static final ExecutorService executionExecutor =
+            Tracers.wrap(Executors.newCachedThreadPool(executionThreads));
```

Comment:
should we use the same type of executor as the okhttp Dispatcher?

Reply:
it is!

Reply:
noted in the Javadoc

Reply:
ah, just had to chase the `Executors` call. thanks.